### PR TITLE
net/frr: add route summarization via area range

### LINKED ISSUE #1148: Feature Request: FRR/OSPF add area rage config to GUI
To summarize routes at ABR I need to set a range statement like this:
```
router ospf
 ospf router-id a.b.c.d
 redistribute connected
 passive-interface vmx0_vlan2000
 passive-interface vmx0_vlan2050
 network 10.1.0.0/27 area 0.0.0.0
 network 10.1.190.0/23 area 0.0.0.11
 area 0.0.0.11 range 10.1.190.0/23
```

With two subnets like this:
```
VLAN 2000 10.1.190.0/24
VLAN 2050 10.1.191.0/27
```

OSPF will inject two separate routes to area 0.0.0.0

With the area..range statement they will be aggregated to a single /23 towards  0.0.0.0

### PR DESCRIPTION
**Pull request by mimugmail**
Closes #1148

### CODE REVIEW

File: net/frr/src/opnsense/mvc/app/controllers/OPNsense/Quagga/forms/dialogEditOSPFNetwork.xml [hunk 1]
```diff
@@ -20,6 +20,12 @@
     <type>text</type>
     <help>Area in wildcard mask style like 0.0.0.0 and no decimal 0</help>
   </field>
+  <field>
+    <id>network.arearange</id>
+    <label>Area Range</label>
+    <type>text</type>
+    <help>To a network to summarize for this area (e.g. 192.168.0.0/23).</help>
```
Comment (fabianfrz) [inline net/frr/src/opnsense/mvc/app/controllers/OPNsense/Quagga/forms/dialogEditOSPFNetwork.xml]: the help string needs improvement.

Comment (mimugmail) [reply net/frr/src/opnsense/mvc/app/controllers/OPNsense/Quagga/forms/dialogEditOSPFNetwork.xml]: Indeed, I flipped the sentence twice and forgot to double check. :)

File: net/frr/src/opnsense/mvc/app/controllers/OPNsense/Quagga/forms/dialogEditOSPFNetwork.xml [hunk 2]
```diff
@@ -20,6 +20,12 @@
     <type>text</type>
     <help>Area in wildcard mask style like 0.0.0.0 and no decimal 0</help>
   </field>
+  <field>
+    <id>network.arearange</id>
+    <label>Area Range</label>
```
Comment (fichtner) [inline net/frr/src/opnsense/mvc/app/controllers/OPNsense/Quagga/forms/dialogEditOSPFNetwork.xml]: Range sounds misleading. Is this a Quagga/FRR term?

Comment (mimugmail) [reply net/frr/src/opnsense/mvc/app/controllers/OPNsense/Quagga/forms/dialogEditOSPFNetwork.xml]: Yep

Comment (fabianfrz) [reply net/frr/src/opnsense/mvc/app/controllers/OPNsense/Quagga/forms/dialogEditOSPFNetwork.xml]: if you look at the config, then probably yes:

```
area {{ network.area }} range {{ network.arearange }}
```

Comment (mimugmail) [reply net/frr/src/opnsense/mvc/app/controllers/OPNsense/Quagga/forms/dialogEditOSPFNetwork.xml]: I think there are more ways to achieve route summarization .. thats why range would fit better

**Comment by fichtner** [comment]
Merged, thanks!

**Pull request merged**